app/Domain/Dice/Dice.h

```diff
@@ -0,0 +1,38 @@
+//
+// Created by Gabriel Noriega on 2018-02-14.
+//
+
+#ifndef SMALLWORLD_DICE_H
+#define SMALLWORLD_DICE_H
+
+#include <iostream>
+#include <vector>
+
+class Dice {
+private:
+    int m_totalRolls{0};
+
+    // Holds the times a face has been rolled (0 = blank, 1 ,2 ,3)
+    std::vector<int> m_rollTracker;
+
+    // Generate random number between 0 - 6
+    int generateRandomNumber();
+
+    // Transform value into 0 - 3 range
+    int transformRoll(int value) const;
+
+public:
+    Dice() : m_rollTracker(4, 0) {}
```

Comment:
Suggested to put the range parameters as arguments to the constructor

Reply:
I disagree, because in that case the user is allowed to create it's own Dice range but the code will still generate numbers 0-3. So, if for any reason the user decided to create a container of size 2, it will break the code. I can modify the code for personalized die sizes with specific number the user want, but I think it is doing more than we need to. Any one that uses this Dice should just be able to:
- roll
- get total number of rolls
- get container of total rolls per number
- display statistics